### DISCUSSION TITLE
Add commands for previous and next note according to the custom sort order (if existing)

### DISCUSSION THREAD

**Discussion by dmarcolongo** [SebastianMC/obsidian-custom-sort#159] [Ideas]
While Obsidian is a file tree have applitcation, there is no command to switch to the previous or next note in the file tree. It would be very useful to have these two commands, and it fits nicely with the custom sort order plugin, which would respect any set custom order.

**Comment by SebastianMC** [comment]
Hi @dmarcolongo 

added this to the list of enhancements.
While this kind of commands are fully independent from custom sort and they operate solely on the internal data model of File Explorer, shipping them as part of custom sort plugin would nicely fit in the scope.

Some iInitial thoughts:

- risky dependency on internal data model of File Explorer - Obsidian team tend to change such internals without any further notice thus breaking compatibility with some dependent plugins. At the same time there is no other data model to rely on
- the logic of such command would have to determine in first step the context: the displayed note (a single one in focus, if any) and then the related File Explorer view (if any).
  - tricky part: there could be more than one File Explorer views (encountered this with custom sort) and none of them active / in focus (the note is in focus) - how to handle that scenario?
  - or maybe the precondition is that a File Explorer view is in focus with some file or folder in tree selected / active?
    - how to interact with File Explorer view and data model to determine it? 
- what would be the result of move-up / move-down command?
  - change in selection / focus in files tree?
  - automatic opening of a note?
    - extremely tricky: notes can be open multiple times in multiple coexisting views - what should be the correct logic?
- aren't there similar plugins already?
- before thinking about any solution need to evaluate this from UX perspective and determine what would be the simplest working method for such a commands

**Comment by SebastianMC** [comment]
I slept on this idea and the conclusion is that it is not feasible.
The effort required for implementation is high and I don't have the time needed for it.
The effort required later on for maintenance is unpredicable, likely high, because of the dependence on Obsidian internals, subject to change between their releases.

Moving this to discussion section - maybe there is someone who can create a plugin with the requested navigation feature? Maybe such a plugin already exists and can be integrated with my custom-sort plugin?

Let's see
